angel_system/data/medical/data_paths.py

```diff
@@ -1,221 +1,121 @@
 
-def grab_data(skill, machine="gyges"):
-    if machine == "gyges":
-        if skill == "m2":
-            return (
-                ptg_root,
-                data_dir,
-                m2_activity_config_fn,
-                m2_activity_gt_dir,
-                None,
-                m2_training_split,
-                m2_obj_dets_dir,
-                m2_obj_config,
-            )
-        elif skill == "m2_lab":
-            return (
-                ptg_root,
-                data_dir,
-                None,
-                None,
-                m2_kitware_data_dir,
-                m2_lab_training_split,
-                None,
-                None,
-            )
-        elif skill == "m2_kitware":
-            return (
-                ptg_root,
-                data_dir,
-                None,
-                None,
-                None,
-                m2_kitware_training_split,
-                None,
-                None,
-            )
+    Parameters:
+        path (str): Root directory path.
+        types (list): List of file types (extensions) to be extracted.
+        recursive (bool, optional): Search for files in subsequent directories if True. Default is False.
 
+    Returns:
+        list: List of file paths with full paths.
+    """
+    files = []
+    if recursive:
+        path = path + "/**/*"
+    for type in types:
+        if recursive:
+            for x in glob(path + type, recursive=True):
+                files.append(os.path.join(path, x))
+        else:
+            for x in glob(path + type):
+                files.append(os.path.join(path, x))
+    return files
+
+class GrabData(object):
+    def __init__(self, yaml_path: str):
+        
+        import yaml
+        with open(yaml_path, 'r') as f:
+            config = yaml.load(f, Loader=yaml.FullLoader)
+        
+        self.data_dir = config['data_dir']
+        self.activity_gt_dir = config['activity_gt_dir']
+        self.objects_dir = config['objects_dir']
+        self.ros_bags_dir = config['ros_bags_dir']
+        
+        self.bbn_data_dir = config['bbn_data_dir']
+        self.bbn_data_root = config['bbn_data_root']
+        
+        self.lab_bbn_data_root = config['lab_bbn_data_root']
+    
+    def grab_data(self, skill: str, data_type: str) -> list:
+        if data_type == "pro":
+            skill_data_root = f"{self.bbn_data_root}/{TASK_TO_NAME[skill]}/Data"
+            videos = os.listdir(skill_data_root)
+            
+            videos_paths = [f"{skill_data_root}/{video}" for video in videos]
+            
+            return videos_paths
+            
+        elif data_type == 'lab':
+            skill_data_root = f"{self.lab_bbn_data_root}/{LAB_TASK_TO_NAME[skill]}/"
+            
+            videos_paths = dictionary_contents(self.skill_data_root, types=['*.mp4'])
+            
+            return videos_paths
+        
         else:
             raise NotImplementedError
+
+
+def grab_data(skill, machine="pro"):
```

Comment:
Remove

Reply:
we can keep this for backward comparability for now. in case someone is using an older commit.

Reply:
I'm good with removing it - It'll be me running things in the short term so I can fix anything that runs into this. It'll confuse me more by leaving it I think.